core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -0,0 +1,99 @@
+package io.grpc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Encloses classes related to the compression and decompression of messages.
+ */
+public final class MessageEncoding {
+
+  /**
+   * Special sentinel codec indicating that no compression should be used.  Users should use
+   * reference equality to see if compression is disabled.
+   */
+  public static final Codec NONE = new Codec() {
+    @Override
+    public InputStream decompress(InputStream is) throws IOException {
+      return is;
+    }
+
+    @Override
+    public String getMessageEncoding() {
+      return null;
```

Comment:
@dgquintas had added "none" to the internal specification doc, not realizing that spec had been superseded. I then created grpc/grpc-common#257 to get "identity" added after discussing with him. Like many fixes to the spec, it has languished. There was one reference to "none" in the compression doc that should have been updated to "identity" but was missed (David just fixed it).

It behaves like Content-Encoding "identity"; "identity" is always implicitly supported and there is no need to send it on the wire.